python/pysrc/kaskada/sources/arrow.py

```diff
@@ -310,6 +309,99 @@ async def add_string(self, csv_string: str | BytesIO) -> None:
             await self._ffi_table.add_pyarrow(batch)
 
 
+class JsonlFile(Source):
```

Comment:
I wonder if we want these to be separate sources (for files and strings) or a single `Jsonl` source thaht has `add_files` and `add_jsonl` (or uses `file://` prefixes to determine the kind of strings).

Unrelated to this PR, but wondering if there would be value (in the future) in implementing this within Rust so we can parse more directly?

Reply:
i think this works for now, but we should talk about maybe cleaning up these methods.

or add a high-level `File(Source):` that does `csv`, `parquet`, `jsonl` based on extension.